transforms/code/ingest_2_parquet/kfp_ray/v1/ingest_2_parquet_wf.py

```diff
@@ -0,0 +1,178 @@
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import kfp.compiler as compiler
+import kfp.components as comp
+import kfp.dsl as dsl
+from kfp_support.workflow_support.utils import (
+    ONE_HOUR_SEC,
+    ONE_WEEK_SEC,
+    ComponentUtils,
+)
+from data_processing.utils import GB
+
+
+# the name of the job script
+EXEC_SCRIPT_NAME: str = "ingest_2_parquet_transform.py"
+
+task_image = "quay.io/dataprep1/data-prep-kit/ingest_2_parquet:0.3.0"
+
+# components
+base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing:0.1.1"
+
+# path to kfp component specifications files
+component_spec_path = "../../../../../kfp/kfp_ray_components/"
+
+# compute execution parameters. Here different tranforms might need different implementations. As
+# a result, instead of creating a component we are creating it in place here.
+compute_exec_params_op = comp.func_to_container_op(
+    func=ComponentUtils.default_compute_execution_params, base_image=base_kfp_image
+)
+# create Ray cluster
+create_ray_op = comp.load_component_from_file(component_spec_path + "createRayClusterComponent.yaml")
+# execute job
+execute_ray_jobs_op = comp.load_component_from_file(component_spec_path + "executeRayJobComponent_multi_s3.yaml")
+# clean up Ray
+cleanup_ray_op = comp.load_component_from_file(component_spec_path + "deleteRayClusterComponent.yaml")
+# Task name is part of the pipeline name, the ray cluster name and the job name in DMF.
+TASK_NAME: str = "ingest_to_parquet"
+PREFIX: str = "ingest_to_parquet"
+
+
+@dsl.pipeline(
+    name=TASK_NAME + "-ray-pipeline",
+    description="Pipeline for converting zip files to parquet",
+)
+def ingest_to_parquet(
+    ray_name: str = "ingest_to_parquet-kfp-ray",  # name of Ray cluster
+    ray_head_options: str = '{"cpu": 1, "memory": 4, "image_pull_secret": "", "image": "' + task_image + '" }',
+    ray_worker_options: str = '{"replicas": 2, "max_replicas": 2, "min_replicas": 2, "cpu": 2, "memory": 4, '
+    '"image_pull_secret": "", "image": "' + task_image + '"}',
+    server_url: str = "http://kuberay-apiserver-service.kuberay.svc.cluster.local:8888",
+    # data access
+    data_s3_config: str = "{'input_folder': 'test/ingest2parquet/input', 'output_folder': 'test/ingest2parquet/output/'}",
+    data_s3_access_secret: str = "s3-secret",
+    data_max_files: int = -1,
+    data_num_samples: int = -1,
+    data_files_to_use: str = "['.zip']",
+    # orchestrator
+    runtime_actor_options: str = f"{{'num_cpus': 0.8, `memory`: {2*GB}}}",
```

Comment:
typo in the quotes of `memory`. It should be `'memory'`.

Reply:
Fixed